routes/index.js

```diff
@@ -150,6 +165,75 @@ module.exports = function(utils, env, nunjucksEnv, appName) {
       app.get( '/slowparse/spec/errors.forbidjs.html', function( req, res ) {
         res.render('/slowparse/spec/errors.forbidjs.html');
       });
+    },
+
+    oauth2Callback: function(req, res, next) {
+      if (req.query.logout) {
+        req.session = null;
+        return res.redirect(301, '/');
+      }
+
+      if (!req.query.code) {
```

Comment:
Yeah, I was conflicted about that as well. The real solution is we follow best practice and implement a good logging system in thimble.